perfkitbenchmarker/linux_benchmarks/specsfs2014_benchmark.py

```diff
@@ -0,0 +1,264 @@
+
+  server = gluster_servers[0] if gluster_servers else None
+  args = [((client, server), {}) for client in clients]
+  vm_util.RunThreaded(_PrepareClient, args)
+
+  _ConfigureSpec(prime_client, clients)
+
+  if not prime_client.is_static:
+    # If the prime client is a static VM, we don't want to clobber their
+    # ssh config or private key and they should have already set up
+    # passwordless ssh between instances.
+    prime_client.AuthenticateVm()
+    with vm_util.NamedTemporaryFile() as tf:
+      tf.write('Host *\n')
+      tf.write('  StrictHostKeyChecking no\n')
+      tf.close()
+      prime_client.PushFile(tf.name, '~/.ssh/config')
+
+
+def _ParseSpecSfsOutput(output, extra_metadata=None):
+  """Returns samples generated from the output of SPEC SFS 2014."""
+  root = xml.etree.ElementTree.fromstring(output)
+  samples = []
+
+  for run in root.findall('run'):
+    metadata = {
+        'benchmark': run.find('benchmark').attrib['name'],
+        'business_metric': run.find('business_metric').text
+    }
+    if extra_metadata:
+      metadata.update(extra_metadata)
+
+    for key in METADATA_KEYS:
+      element = run.find('metric[@name="%s"]' % key)
+      units = element.attrib.get('units')
+      label = (key + ' (%s)' % units) if units else key
+      metadata[label] = element.text
+
+    for metric in run.findall('metric'):
+      name = metric.attrib['name']
+      if name in PUBLISHED_METRICS:
+        samples.append(sample.Sample(name,
+                                     float(metric.text),
+                                     metric.attrib.get('units', ''),
+                                     metadata))
+  return samples
+
+
+def Run(benchmark_spec):
+  """Run SPEC SFS 2014.
+
+  Args:
+    benchmark_spec: The benchmark specification. Contains all data that is
+        required to run the benchmark.
+
+  Returns:
+    A list of sample.Sample objects.
+  """
+  prime_client = benchmark_spec.vm_groups['clients'][0]
+
+  prime_client.RemoteCommand('cd %s && ./sfsmanager -r sfs_rc' % SPEC_DIR)
+  results_file = posixpath.join(SPEC_DIR, 'results', 'sfssum_sfs2014.xml')
+  output, _ = prime_client.RemoteCommand('cat %s' % results_file)
+
+  gluster_metadata = {
+      'gluster_stripes': FLAGS.gluster_stripes,
+      'gluster_replicas': FLAGS.gluster_replicas
+  }
+
+  return _ParseSpecSfsOutput(output, extra_metadata=gluster_metadata)
+
+
+def Cleanup(benchmark_spec):
+  """Cleanup SPEC SFS 2014.
+
+  Args:
+    benchmark_spec: The benchmark specification. Contains all data that is
+        required to run the benchmark.
+  """
+  clients = benchmark_spec.vm_groups['clients']
+  gluster_servers = benchmark_spec.vm_groups['gluster_servers']
+
+  for client in clients:
+    client.RemoteCommand('sudo umount %s' % MOUNT_POINT)
+    client.RemoteCommand(
+        'rm %s && sudo rm -rf %s' % (SPEC_SFS_2014_ISO, SPEC_DIR))
+
+  gluster.DeleteVolume(gluster_servers[0], VOLUME_NAME)
```

Comment:
It looks like there are other places in code where gluster_servers can be empty. I'm not sure what it would mean to run without them, but make sure it's consistent.